### PR TITLE
Native SRT protocol support

### PR DESCRIPTION
**Pull request by J-Rogmann**
### Description

Adding native support for the Secure Reliable Transport ([SRT](https://github.com/Haivision/srt)) protocol to OBS Studio instead of using the ffmpeg implementation of SRT can unlock additional features and make it easier to use.

### Motivation and Context

The SRT protocol enables low-latency point-to-point and point-to-multipoint streaming. It also plays an increasing role for ingest into CDNs. Recently AWS Elemental MediaConnect enabled SRT ingest, and we are sure that many CDNs and streaming-platforms will follow.

### [Link to RFC](https://github.com/J-Rogmann/rfcs/blob/native-srt-support/text/0044-native-srt-protocol-support.md)

### CODE REVIEW

**Comment by dodgepong** [comment]
Can you include more details about what implementation might look like in OBS? Including proposed UI/UX changes. Otherwise this is just a glorified feature request.

File: text/0044-native-srt-protocol-support.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+- SRT can work with bi-directional streaming, allowing workflows like talkback or bi-directional audio/video communication.
+- SRT supports 128/256-bit AES encryption.
+- SRT is currently in a standardisation process with the IETF. (See the [RFC submitted](https://datatracker.ietf.org/doc/html/draft-sharabayko-srt-00)
+- SRT enables easy firewall traversal with Caller, Listener, and Rendezvous modes.
+
+## Benefits of a Native SRT Implementation in OBS Studio
+
+Currently OBS Studio supports SRT through the underlying ffmpeg implementation. However, this approach has drawbacks in terms of both how the protocol can be used and which of its features are supported.
+
+- All SRT options have to be specified in a command line to be passed to ffmpeg. This is complicated for inexperienced users, and beginners might be overwhelmed by the number of available [SRT API socket options](https://github.com/Haivision/srt/blob/master/docs/API/API-socket-options.md#list-of-options). A UI integration would improve usability and adoption.
```
Comment (kkartaltepe) [inline text/0044-native-srt-protocol-support.md:27]: This is not true, we do not need a native implementation to add options to OBS.

File: text/0044-native-srt-protocol-support.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- SRT statistics: One of the biggest benefits for streaming operators is the [rich live statistics](https://github.com/Haivision/srt/blob/master/docs/API/statistics.md) which SRT provides. Having visibility into a stream's  bandwidth usage, RTT values, packet loss rate, buffer levels and re-transmission attempts makes it easier to troubleshoot problems. SRT can produce graphs to display data issues, making them more easily identifiable. These data can be used to prove stability or other issues with clients or ISPs. This data can also be used in applications to control the actual video encoding bitrate according to network performance, like Haivision does with [Network Adaptive Encoding](https://www.haivision.com/resources/streaming-video-definitions/network-adaptive-encoding/). 
+- SRT connection bonding: It is sometimes necessary to explicitly specify the NICs being used for streaming. Instead of looking up NIC names and adding them to the command line for ffmpeg, having a drop-down menu in the UI auto populated with available NICs would be a great help to users.
```
Comment (kkartaltepe) [inline text/0044-native-srt-protocol-support.md:30]: This is just the first point reiterated.

**Comment by Fenrirthviti** [comment]
There's an open PR for this, so this RFC is no longer necessary. As stated by dodgepong, this is effectively a feature request anyway, and should probably not have been submitted as an RFC.

https://github.com/obsproject/obs-studio/pull/6134

**Comment by pkviet** [comment]
@J-Rogmann odd as it may seem, i wasn't aware of this RFC. If you want to contribute to the PR, by adding features, I'll be very happy to talk. I can be reached on video-dev slack (subscribed to the srt channel) and on obs discord. I was in contact with Marc and others from your company a few years back to debug our srt support in obs.
I copied the implementation from the ffmpeg one in my PR. But added more stats so this is one of your bullet points.
As to better UI for passing options, I plan to add that but this will be after the core native support has been merged.
For the auto-reconnect, I wasn't aware of the mechanism.
Bonding would be nice too. 
I'm personally interested in having some dynamic bitrate. I think it's supported in some haivision hw encoders. But it's not implemented in srt-live-transmit.

**Comment by pkviet** [comment]
@J-Rogmann i'm not subscribed to this repo so didn't see the pr. I'll look into your suggestions and if i have questions i'll contact you.

**Pull request closed without merging**